Add caching policy to github action, and filter built files

Razboy20 opened the pull request:
Should be fine to merge into other branches; also only applies to the build release action.

coderbot16 commented on .github/workflows/build-release.yml:
```diff
@@ -12,22 +12,41 @@ jobs:
+            ~/.gradle/loom-cache
+            ~/.gradle/wrapper
+            ~/.m2/repository
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
```
I assume it's intended that gradle.properties isn't included here?

Razboy20 replied:
Yes, although it doesn't really matter much tbh. Gradle is pretty smart with its cachings, and custom keys per build.gradle are probably not even needed, but just in case. (This is also the shown example in the cache docs)

coderbot16 commented on .github/workflows/build-release.yml:
```diff
@@ -12,22 +12,41 @@ jobs:
         run: ./gradlew publishToMavenLocal
       - name: Build initial Sodium artifacts
         run: cd ./sodium-fabric && ./gradlew publishToMavenLocal && cd ..
-      - name: List contents of maven local repository
-        run: ls -lR /home/runner/.m2/repository
```
Personally I don't see a huge issue with keeping these, if there are issues with dependencies not being found it's helpful to see the content of mavenlocal.

coderbot16 commented on .github/workflows/build-release.yml:
```diff
@@ -12,22 +12,41 @@ jobs:
         with:
           name: build-artifacts
-          path: build/libs
+          path: build/libs/*[0-9].jar
```
If possible it would be nice to annotate the intent of this with a comment. I assume this is trying to capture the "-build.*" jars without the "-dev" or "-sources" jars? Maybe a more explicit pattern would be nice (or if there's an explicit way to exclude -dev or -sources with the regex)

Razboy20 replied:
Sadly, the path does not support regex, but rather @action/glob syntax which is weird. I shall add a comment describing it however to make it more clear what the line does.

coderbot16 commented on .github/workflows/build-release.yml:
```diff
@@ -12,22 +12,41 @@ jobs:
-        uses: actions/upload-artifact@v1
+        uses: actions/upload-artifact@v2
         with:
           name: build-artifacts
```
While we're at it, is there any way to make the name of the build artifacts zip be dependent on the branch name? That would be helpful, just curious.

coderbot16 reviewed (changes requested):
Thanks! This should be helpful. I have a few comments though.

Razboy20 commented:
Changed everything that you asked, and also added comments to explain some of the more confusing segments of the action. Artifact names are now `iris-artifacts-<branch name>`, where 1.17/dev would become `iris-artifacts-1.17-dev`

coderbot16 reviewed (approved):
Thanks! This should definitely help speed up the actions builds, and the naming of the build artifacts zip should help clear things up as well.
